### ISSUE TITLE
Error when i testing the demo_cityscapes and demo_NYUv2

### ISSUE THREAD

**Issue by xjlnuc** [aimerykong/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop#10]
Hi aimerykong:
The error are as follows
错误使用 dagnn.Layer/load (line 200)
No property `bnorm_moment_type_trn` for a layer of type `dagnn.BatchNorm`.

出错 dagnn.DagNN.loadobj (line 27)
    block.load(struct(s.layers(l).block)) ;

出错 demo_Cityscapes (line 82)
netMat = dagnn.DagNN.loadobj(netMat);

错误使用 dagnn.Layer/load (line 200)
No property `noise_param_idx` for a layer of type `dagnn.BatchNorm`.

出错 dagnn.DagNN.loadobj (line 28)
      block.load(struct(s.layers(l).block)) ;

出错 demo_NYUv2 (line 40)
netMat = dagnn.DagNN.loadobj(netMat);

**Comment by aimerykong** [comment]
Hi, it seems you use a different matconvnet toolbox? Do you download and
use the official version? I add new layers, so please make sure you are
using the one I uploaded in github. Thanks!

**Comment by xjlnuc** [comment]
Hi aimerykong:
Thank you very much for the quick reply，you are right,i have use a different matconvnet toolbox,but when i use the official version,I can compile successfully,and when i use your matconvnet,there are many problems when i compiling for gpu,the problem is

F:/personal file2/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop-master/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop-master/matconvnet/matlab/src/bits/impl/bilinearsampler_gpu.cu(247): warning: variable "backward" was declared but never referenced 
Detected during:instantiation of "vl::ErrorCode forward_backward<type,backwardData,backwardGrid>(vl::Context &, type *, type *, type *, const type *, const type *, const type *, size_t, size_t, size_t, size_t, size_t, size_t, size_t) [with type=float, backwardData=true, backwardGrid=false]"  
(352): here 
            instantiation of "vl::ErrorCode vl::impl::bilinearsampler<vl::VLDT_GPU, type>::backward(vl::Context &, type *, type *, const type *, const type *, const type *, size_t, size_t, size_t, size_t, size_t, size_t, size_t) [with type=float]"  

 i don’t konw what happend 
And,when test the cpu and gpu,the result are as follows 
![image](https://user-images.githubusercontent.com/30426708/30529154-0d8c5b5e-9c6d-11e7-8ff7-1d7e2024e0e2.png)
![image](https://user-images.githubusercontent.com/30426708/30529184-56926988-9c6d-11e7-8e84-481e10092ab4.png)
I think I have not compiled successfully,because when i am running the demo_cityspace on this basis,there are errors as follows:
![image](https://user-images.githubusercontent.com/30426708/30529194-7dcfed68-9c6d-11e7-82a6-b4713257a6d4.png)
I have use the win10+matlab2017a+vs2015+cuda8.0+cudnn5.1,Is it because I am using the wrong version?Can you tell me what version you are using?I would be grateful.

**Comment by aimerykong** [comment]
Hi, can you compile it without testing. besides writing new functions for
the new layers, I didn't add them in the testing scripts. How about
compiling using CPU only? If that works, compile using GPU without cudnn.

**Comment by xjlnuc** [comment]
Hi,i can successfully compile cpu,and i have compiled  GPU without cudnn,but there is still have the same result when i run the demo_cityspaces.I will try win7 to see if it can be successful.
       The following is my problem in the compile of the GPU,  and I think the code could be the cause of the error：
1.
d:\recurrent-scene-parsing-with-perspective-understanding-in-the-loop-master\matconvnet\matlab\src\bits/impl/tinythread.h(654): warning: extra ";" ignored 
 
D:/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop-master/matconvnet/matlab/src/vl_imreadjpeg.cu(1329): warning: statement is unreachable 

2.
D:/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop-master/matconvnet/matlab/src/bits/impl/bilinearsampler_gpu.cu(247): warning: variable "backward" was declared but never referenced 

3.
D:/Recurrent-Scene-Parsing-with-Perspective-Understanding-in-the-loop-master/matconvnet/matlab/src/bits/impl/bilinearsampler_gpu.cu(247): warning: variable "backward" was declared but never referenced 
          detected during: 
            instantiation of "vl::ErrorCode forward_backward<type,backwardData,backwardGrid>(vl::Context &, type *, type *, type *, const type *, const type *, const type *, size_t, size_t, size_t, size_t, size_t, size_t, size_t) [with type=float, backwardData=false, backwardGrid=false]"  
(325): here 
            instantiation of "vl::ErrorCode vl::impl::bilinearsampler<vl::VLDT_GPU, type>::forward(vl::Context &, type *, const type *, const type *, size_t, size_t, size_t, size_t, size_t, size_t, size_t) [with type=float]"  
(364): here 

        Thank you very much!

**Comment by aimerykong** [comment]
These are not error. These are just warnings. So you have successfully
compiled with GPU.
